src/Okta.Idx.Sdk/IdxClient.cs

```diff
@@ -1510,5 +1536,240 @@ public async Task<TResponse> SendAsync<TResponse>(HttpRequest request, HttpVerb
+                CanSkip = selectAuthenticatorResponse.ContainsRemediationOption(RemediationType.Skip),
+            };
+
+            return authenticationResponse;
+        }
+
+        /// <inheritdoc/>
+        public async Task<AuthenticationResponse> SelectEnrollAuthenticatorAsync(EnrollOktaVerifyAuthenticatorOptions enrollAuthenticatorOptions, IIdxContext idxContext, CancellationToken cancellationToken = default)
+        {
+            var introspectResponse = await IntrospectAsync(idxContext, cancellationToken);
+
+            if (!introspectResponse.ContainsRemediationOption(RemediationType.SelectEnrollmentChannel))
+            {
+                throw new UnexpectedRemediationException(
+                    new List<string>
+                    {
+                        RemediationType.SelectEnrollmentChannel,
+                    }, introspectResponse);
+            }
+
+            var idxRequestPayload = new IdxRequestPayload();
+            idxRequestPayload.SetProperty("authenticator", new { channel = enrollAuthenticatorOptions.Channel, id = enrollAuthenticatorOptions.AuthenticatorId });
+            idxRequestPayload.SetProperty("stateHandle", introspectResponse.StateHandle);
+
+            var selectEnrollmentChannelResponse = await introspectResponse.ProceedWithRemediationOptionAsync(
+                RemediationType.SelectEnrollmentChannel, idxRequestPayload, cancellationToken);
+
+            var authenticationResponse = new AuthenticationResponse
+            {
+                IdxContext = idxContext,
+                AuthenticationStatus = AuthenticationStatus.AwaitingAuthenticatorEnrollmentData,
+                CurrentAuthenticator = IdxResponseHelper.ConvertToAuthenticator(selectEnrollmentChannelResponse.Authenticators?.Value, selectEnrollmentChannelResponse.CurrentAuthenticator?.Value),
+                CanSkip = selectEnrollmentChannelResponse.ContainsRemediationOption(RemediationType.Skip),
+            };
+
+            return authenticationResponse;
+        }
+
+        /// <inheritdoc/>
+        public async Task<AuthenticationResponse> EnrollAuthenticatorAsync(EnrollOktaVerifyAuthenticatorOptions oktaVerifyAuthenticatorOptions, IIdxContext idxContext, CancellationToken cancellationToken = default)
+        {
+            var introspectResponse = await IntrospectAsync(idxContext, cancellationToken);
+            if (!introspectResponse.ContainsRemediationOption(RemediationType.EnrollmentChannelData))
+            {
+                throw new UnexpectedRemediationException(
+                    new List<string>
+                    {
+                        RemediationType.EnrollmentChannelData,
+                    }, introspectResponse);
+            }
+
+            var idxRequestPayload = new IdxRequestPayload { StateHandle = introspectResponse.StateHandle };
+
+            switch (oktaVerifyAuthenticatorOptions.Channel)
+            {
+                case "email":

```

Comment:
Nit: Use AuthenticatorMethodType instead of hardcoding these values.

Reply:
That results in a compiler error.